slither/slithir/convert.py

```diff
@@ -385,6 +385,70 @@ def integrate_value_gas(result: List[Operation]) -> List[Operation]:
 ###################################################################################
 
 
+def get_declared_param_names(ins: Call) -> Optional[List[str]]:
+    """
+    Given a call operation, return the list of parameter names, in the order
+    listed in the function declaration.
+    #### Parameters
+    ins -
+        The call instruction
+    #### Possible Returns
+    List[str] -
+        A list of the parameters in declaration order
+    None -
+        Workaround: Unable to obtain list of parameters in declaration order
+    """
+    if isinstance(ins, NewStructure):
+        return [x.name for x in ins.structure.elems_ordered if not isinstance(x.type, MappingType)]
+    if isinstance(ins, NewContract):
+        return [p.name for p in ins.contract.constructor.parameters]
+    if isinstance(ins, (LowLevelCall, NewElementaryType, NewArray)):
+        # named arguments are incompatible with these call forms
+        assert False
```

Comment:
Can we update the function signature to only be subclasses of call which are accepted and remove this?  We can add an assertion like `assert isinstance(ins, NewStructure, NewContract... )`, too

Reply:
It might be better to add `assert not isinstance(LowLevelCall, NewElementaryType, NewArray)` at the top of the `get_declared_param_names` function.

We could also make the `assert False` case return `None` instead. Of course, it would be a hassle to get test coverage for this.